File: lib/log.js
import { appendFileSync, mkdirSync } from 'fs';
import { join } from 'path';
import { ROOT } from './state.js';

const LOGS_DIR = join(ROOT, 'logs');
mkdirSync(LOGS_DIR, { recursive: true });

function ts() {
  return new Date().toISOString();
}

function logLine(level, msg, data) {
  const date = new Date().toISOString().split('T')[0];
  const line = `[${ts()}] ${level}: ${msg}${data ? ' ' + JSON.stringify(data) : ''}\n`;
  process.stderr.write(line);
  try {
    appendFileSync(join(LOGS_DIR, `${date}.log`), line);
  } catch { /* ignore */ }
}

export const log = {
  info: (msg, data) => logLine('INFO', msg, data),
  warn: (msg, data) => logLine('WARN', msg, data),
  error: (msg, data) => logLine('ERROR', msg, data)
};

File: internal/log/log.go
package log

import (
	"fmt"
	"os"
	"path/filepath"
	"time"

	"github.com/Pernek-Enterprises/dispatch/internal/config"
)

func ts() string {
	return time.Now().UTC().Format(time.RFC3339)
}

func write(level, msg string) {
	line := fmt.Sprintf("[%s] %s: %s\n", ts(), level, msg)
	fmt.Fprint(os.Stderr, line)

	date := time.Now().Format("2006-01-02")
	logDir := filepath.Join(config.Root, "logs")
	os.MkdirAll(logDir, 0755)
	f, err := os.OpenFile(filepath.Join(logDir, date+".log"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
	if err == nil {
		f.WriteString(line)
		f.Close()
	}
}

func Info(msg string, args ...interface{})  { write("INFO", fmt.Sprintf(msg, args...)) }
func Warn(msg string, args ...interface{})  { write("WARN", fmt.Sprintf(msg, args...)) }
func Error(msg string, args ...interface{}) { write("ERROR", fmt.Sprintf(msg, args...)) }
